packages/core/src/PasswordInput/PasswordInput.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import {
+  Visibility,
+  VisibilityOff,
+  Check,
+  Success,
+  SuccessOutline,
+} from 'pcln-icons'
+import {
+  Flex,
+  Text,
+  Input,
+  Label,
+  IconField,
+  Badge,
+  ProgressBar,
+  IconButton,
+} from '../../src'
+
+const CustomText = styled(Text)`
+  white-space: nowrap;
+`
+
+const maxProgressBarLength = 4
+
+function PasswordInput({
+  label,
+  hasProgressBar,
+  progressBarSteps,
+  regexChecks,
+  value,
+  onChange,
+}) {
+  const [showPassword, setShowPassword] = useState(false)
+  const [passedChecks, setPassedChecks] = useState([])
+
+  const changeVisibility = () => setShowPassword(!showPassword)
+
+  const handleChange = (event) => {
+    const currentInput = event.target.value
+    const passedChecks = []
+
+    regexChecks.reduce(
+      (acc, element, index) =>
+        element.regex.test(currentInput) && passedChecks.push(index),
+      passedChecks
+    )
+
+    setPassedChecks(passedChecks)
+
+    const isValid = passedChecks.length === regexChecks.length
+    onChange && onChange({ isValid, value: currentInput })
+  }
+
+  const inputType = showPassword ? 'text' : 'password'
+  const VisibilityIcon = showPassword ? VisibilityOff : Visibility
+
+  const strengthLevel = Math.max(
+    Math.floor(
+      (maxProgressBarLength / regexChecks.length) * passedChecks.length
+    ),
+    1
+  )
+  const currentStep = progressBarSteps[strengthLevel - 1]
+  const showCheckIcon = strengthLevel === maxProgressBarLength
+
+  return (
+    <Flex flexDirection='column'>
+      <Label fontSize={0} mb={2}>
+        {label}
+      </Label>
+      <IconField>
+        <Input
+          type={inputType}
+          value={value}
+          onChange={handleChange}
+          data-testid='input-field'
+        />
```

Comment:
It'd be really cool if we supported suggested passwords so users can quick generate a random password and have their browser save it. I think it's as simple as adding `autocomplete='new-password'` however we'd want to allow for devs to set this themselves since they might want to use this for a current password input. TLDR: could you add a `autocomplete` prop and attach it to the input please and thanks 😄